sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpRequestMetadata.java

```diff
@@ -107,42 +108,27 @@ public HttpRequestMetadata setEagerlyConvertHeaders(boolean eagerlyConvertHeader
     }
 
     /**
-     * Whether the response body should be eagerly read.
-     *
-     * @return Whether the response body should be eagerly read.
-     */
-    public boolean isEagerlyReadResponse() {
-        return eagerlyReadResponse;
-    }
-
-    /**
-     * Sets whether the response body should be eagerly read.
+     * Gets a {@link ResponseBodyHandling value} indicating how the body of the resulting HTTP response should be
```

Comment:
nit: don't use the renamed link here

`Get a value indicating ...`

This isn't very readable at first glance because value can be anything, use the name of the class to make it clearer.

`Get a ResponseBodyHandling indacting ...`

or better change the Javadoc to explain without needing a link

```
* Gets the configuration indicating how the body of an HTTP response should be handled.
* <p>
* For more information about the options for handling an HTTP response body see {@link ResponseBodyHandling}.
```

And don't forget to include information about the default handling